### ISSUE TITLE
How do I specify the SwaggerRoute prefix?

### ISSUE THREAD

**Issue by jonesmat** [Jusas/NSwag.AzureFunctionsV2#4]
It looks like all routes are prefixed with "api/", how can I customize this?  I'd to be able to specify the version of the api (i.e. "api/v1/")

Thanks!

**Comment by Jusas** [comment]
Hi @jonesmat 
The default route for Azure Functions is /api/<FunctionName>. This is however overrideable via host.json according to https://docs.microsoft.com/en-us/sandbox/functions-recipes/routes?tabs=csharp
Additionally custom routes can be defined in the HttpTrigger attribute.
So you're pretty much confined to whatever that syntax supports, routing in Functions is a bit more limited.

However there's one detail that just occurred to me now that I checked the code: while you can change the prefix in host.json, the Swagger Generator code assumes it to always be the default /api/. This is something I have to investigate if I can in some nice way detect the route prefix, because right now if you're using a custom route prefix in host.json the generated Swagger document will have broken routes.

**Comment by RicoSuter** [comment]
@jusas is your generator also based on the api explorer matadata? If you then it should just work or it is a bug in the function runtime

**Comment by Jusas** [comment]
> 
> 
> @Jusas is your generator also based on the api explorer matadata? If you then it should just work or it is a bug in the function runtime

No, as far as I know, API explorer is not available in Azure Functions. It's been a while since I checked though.

Edit: Probably not, since Functions aren't Controllers at all, and don't really follow any ASP.NET conventions.

**Comment by jonesmat** [comment]
@Jusas Thanks for the response.  Do you think a short term solution would be to allow a setting to be passed to the generator for the route prefix?  Not ideal, but better than it generating broken routes until a more elegant solution can be implemented.

**Comment by Jusas** [comment]
Yes, that can definitely be done. I'll take a look at it today and also see
if a more elegant solution can be devised, but I think a setting would in
fact be perfectly fine.

**Comment by Jusas** [comment]
@jonesmat I just released v1.1.2 that has support for setting the route prefix. So, after setting it in your host.json, you'll need to set it in your AzureFunctionsV2ToSwaggerGeneratorSettings, so the basic usage goes like:
```
var settings = new AzureFunctionsV2ToSwaggerGeneratorSettings();

...

// to replace 'api' with 'my-route-prefix'
settings.RoutePrefix = "my-route-prefix";
// or to remove the 'api' from the front
settings.RoutePrefix = "";

var funcClasses = new[] { typeof(MyFunctions) };
var generator = new AzureFunctionsV2ToSwaggerGenerator(settings);
var document = await generator.GenerateForAzureFunctionClassesAsync(funcClasses, null);

...
```

If RoutePrefix is not set, it defaults to the Functions default, `'api'`

**Comment by jonesmat** [comment]
@Jusas Great stuff!  Justed tested it out and it works exactly as I'd expect.
